java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -3376,16 +3376,27 @@ void testFixedWidthCast() {
   void testCastBigDecimalToString() {
     BigDecimal[] bigValues = {new BigDecimal("923121331938210123.321"),
         new BigDecimal("9223372036854775808.191"),
+        new BigDecimal("-9.223"),
+        new BigDecimal("0.000"),
         new BigDecimal("9328323982309091029831.002")
     };
 
     try (ColumnVector cv = ColumnVector.fromDecimals(bigValues);
          ColumnVector values = cv.castTo(DType.STRING);
          ColumnVector expected = ColumnVector.fromStrings("923121331938210123.321",
              "9223372036854775808.191",
-             "9328323982309091029831.002")) {
+             "-9.223",
+             "0.000",
+            "9328323982309091029831.002")) {
       assertColumnsAreEqual(expected, values);
     }
+
+    BigDecimal[] bigValues0 = {new BigDecimal("992983283728193827182918744829283742232")};
+    try {
+      ColumnVector cv = ColumnVector.fromDecimals(bigValues0);
```

Comment:
Should we make sure that this didn't leak?

Reply:
Obviously we should!